src/dawn/Optimizer/OptimizerContext.cpp

```diff
@@ -654,4 +644,44 @@ const Options& OptimizerContext::getOptions() const { return options_; }
 
 Options& OptimizerContext::getOptions() { return options_; }
 
+void OptimizerContext::fillIIR() {
+  for(const auto& stencil : SIR_->Stencils)
```

Comment:
Since it is permitted (and also happens) to have `SIR_ = nullptr`, it would be safer to `DAWN_ASSERT` here if `fillIIR()` is called with `SIR_` non-null. Also `getSIR()` should be made safe.

Reply:
I'm wondering how this happens, but added assert.